### PR TITLE
Vue webpack

### PR DESCRIPTION
**Pull request by mcallegari10**
## Summary

- Changed Gulp for Webpack 4, still using Pug and SASS
- Add Vue base configuration

To be added: SFC configuration

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,177 +1,6 @@
...
-    "strict": 0,                                                  // http://eslint.org/docs/rules/strict
-    "use-isnan": 2,                                               // http://eslint.org/docs/rules/use-isnan
-    "valid-jsdoc": 0,                                             // http://eslint.org/docs/rules/valid-jsdoc
-    "valid-typeof": 2,                                            // http://eslint.org/docs/rules/valid-typeof
-    "vars-on-top": 0,                                             // http://eslint.org/docs/rules/vars-on-top
-    "wrap-iife": [2, "inside"],                                   // http://eslint.org/docs/rules/wrap-iife
-    "wrap-regex": 0,                                              // http://eslint.org/docs/rules/wrap-regex
-    "yoda": 2                                                     // http://eslint.org/docs/rules/yoda
-  }
+  "extends": "plugin:vue/essential"
```
Comment (FrankIglesias) [inline .eslintrc]: :👏

File: .woloxci/Dockerfile [hunk 2]
```diff
@@ -14,12 +14,10 @@ ENV NODE_PATH=/install/node_modules
 # A wildcard is used to ensure both package.json AND package-lock.json are copied
 # where available (npm@5+)
 COPY package*.json /install/
-COPY yarn* /install/
 
 WORKDIR /install
 
-RUN yarn install
-RUN npm install -g gulp
+RUN npm install
```
Comment (FrankIglesias) [inline .woloxci/Dockerfile]: npm i

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  },
+  resolve: {
+    extensions: ['.js'],
+    alias: {
+      vue: 'vue/dist/vue.js'
+    }
+  },
+  module: {
+    rules: [
+      {
```
Comment (FrankIglesias) [inline webpack.config.js]: Define a rule for vue files

File: src/views/index/index.pug [hunk 4]
```diff
@@ -0,0 +1,14 @@
+doctype html
+html(lang='en')
+  head
+    title title here
+    meta(charset='UTF-8')
+    meta(name='viewport' content='width=device-width, initial-scale=1')
+    link(rel='stylesheet' href='styles.index.css')
+  body
+    #app
+      .center
```
Comment (FrankIglesias) [inline src/views/index/index.pug]: Add the v-cloak

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  },
+  target: 'web',
+  mode: 'development',
+  // TODO: add more options to make the build process more readable and enable HMR for nonJS files
+  devServer: {
+    contentBase: path.resolve(__dirname, 'build'),
+    compress: true,
+    port: 3000,
+    hot: true,
+    https: true
```
Comment (FrankIglesias) [inline webpack.config.js]: Disable this because of cors issue

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+                  flexbox: 'no-2009'
+                })
+              ]
+            }
+          },
+        ]
+      },
+      {
+        test: /\.(jpg|png|gif|svg)$/,
+        use: 'file-loader?name=assets/[name].[ext]'
```
Comment (FrankIglesias) [inline webpack.config.js]: Add font file types

File: webpack.config.prod.js [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+                  flexbox: 'no-2009'
+                })
+              ]
+            }
+          }
+        ]
+      },
+      {
+        test: /\.(jpg|png|gif|svg)$/,
+        use: [
```
Comment (FrankIglesias) [inline webpack.config.prod.js]: Same

**Comment by FrankIglesias** [comment]
It'd be nice to add a module resolver config.

File: README.md [hunk 8]
```diff
@@ -17,43 +17,21 @@ Get the latest version of node from the [official website](https://nodejs.org/)
...
+Run `npm i` from rootpath of the project.
 
 ## Development
 
-#### Environments
-By default, the environment will be **development**, but you can easily change it using the **env** param in your gulp tasks: ```gulp --env production```
+#### Run the project
+After doing `npm i`,  execute `npm start`. This will run the project instance in:
+```
+  https://localhost:3000/
```
Comment (FrankIglesias) [inline README.md]: http

**Comment by SKOLZ** [comment]
We'll finish reviewing this pull request after finishing this bootstrap implementation

**Pull request merged**